src/sentry/debug_files/artifact_bundles.py

```diff
@@ -139,3 +159,220 @@ def _index_urls_in_bundle(
+                    ProjectArtifactBundle.objects.filter(
+                        artifact_bundle_id=artifact_bundle_id, date_added__lte=threshold_date
+                    ).update(date_added=now)
+                    ReleaseArtifactBundle.objects.filter(
+                        artifact_bundle_id=artifact_bundle_id, date_added__lte=threshold_date
+                    ).update(date_added=now)
+                    DebugIdArtifactBundle.objects.filter(
+                        artifact_bundle_id=artifact_bundle_id, date_added__lte=threshold_date
+                    ).update(date_added=now)
+                    ArtifactBundleIndex.objects.filter(
+                        artifact_bundle_id=artifact_bundle_id, date_added__lte=threshold_date
+                    ).update(date_added=now)
+
+            # If the transaction succeeded, and we did actually modify some rows, we want to track the metric.
+            if updated_rows_count > 0:
+                metrics.incr("artifact_bundle_renewal.were_renewed")
+
+
+# ===== Querying of Artifact Bundles =====
+
+
+def _maybe_renew_and_return_bundles(
+    bundles: Dict[int, Tuple[datetime, str]]
+) -> List[Tuple[int, str]]:
+    maybe_renew_artifact_bundles(
+        {id: date_added for id, (date_added, _resolved) in bundles.items()}
+    )
+
+    return [(id, resolved) for id, (_date_added, resolved) in bundles.items()]
+
+
+def query_artifact_bundles_containing_file(
+    project: Project,
+    release: str,
+    dist: str,
+    url: str,
+    debug_id: str | None,
+) -> List[Tuple[int, str]]:
+    """
+    This looks up the artifact bundles that satisfy the query consisting of
+    `release`, `dist`, `url` and `debug_id`.
+
+    This function should ideally return a single bundle containing the file matching
+    the query. However it can also return more than a single bundle in case no
+    complete index is available, in which case the N most recent bundles will be
+    returned under the assumption that one of those may contain the file.
+
+    Along the bundles `id`, it also returns the most-precise method the bundles
+    was resolved with.
+    """
+
+    if debug_id:
+        bundles = get_artifact_bundles_containing_debug_id(project, debug_id)
+        if bundles:
+            return _maybe_renew_and_return_bundles(
+                {id: (date_added, "debug-id") for id, date_added in bundles}
+            )
+
+    total_release_bundles, not_indexed = get_release_indexing_state(project, release, dist)
+
+    if not total_release_bundles:
+        return []
+
+    # If all the bundles for this release are full indexed, we will only query
+    # the url index.
+    # Otherwise, if we are below the threshold, or only partially indexed, we
+    # want to return the N most recent bundles associated with the release,
+    # under the assumption that one of those should ideally contain the file we
+    # are looking for.
+    partially_indexed = total_release_bundles <= INDEXING_THRESHOLD
```

Comment:
I would call this condition `indexes_not_available` or call it `indexes_available` and flip all the logic below. The current name is pretty confusing.